mailbox/opensearch/src/main/java/org/apache/james/mailbox/opensearch/events/OpenSearchListeningMessageSearchIndex.java

```diff
@@ -131,7 +142,61 @@ public EnumSet<SearchCapabilities> getSupportedCapabilities(EnumSet<MessageCapab
             SearchCapabilities.AttachmentFileName,
             SearchCapabilities.PartialEmailMatch);
     }
-    
+
+    @Override
+    public Mono<Void> reactiveEvent(Event event) {
+        MailboxSession systemSession = sessionProvider.createSystemSession(event.getUsername());
+        return handleMailboxEvent(event, systemSession, (MailboxEvents.MailboxEvent) event)
+            .then(Mono.fromRunnable(() -> factory.endProcessingRequest(systemSession)));
+    }
+
+    private Mono<Void> handleMailboxEvent(Event event, MailboxSession session, MailboxEvents.MailboxEvent mailboxEvent) {
+        MailboxId mailboxId = mailboxEvent.getMailboxId();
+
+        if (event instanceof MailboxEvents.Added) {
+            MailboxEvents.Added addedEvent = (MailboxEvents.Added) event;
+
+            if (addedEvent.isMoved()) {
+                MailboxId movedFromMailboxId = addedEvent.movedFromMailboxId().get();
+                return Flux.fromIterable(addedEvent.getUids())
+                    .concatMap(messageUid -> addByIndexedDocument(movedFromMailboxId, addedEvent.getMailboxId(), messageUid))
+                    .onErrorResume(OpenSearchNotFoundException.class,
+                        e -> addByFetchMailboxMessage((MailboxEvents.Added) event, session, mailboxId))
+                    .then();
+            } else {
+                return addByFetchMailboxMessage((MailboxEvents.Added) event, session, mailboxId);
+            }
+        } else if (event instanceof MailboxEvents.Expunged) {
+            MailboxEvents.Expunged expunged = (MailboxEvents.Expunged) event;
+
+            return delete(session, mailboxId, expunged.getUids());
```

Comment:
because we still need to create a new document and remove an old document. 
=> The ignore delete when `isMoved` is not the correct way